domain/controllernode/state/state.go

```diff
@@ -235,3 +235,36 @@ UPDATE SET version = excluded.version, architecture_id = excluded.architecture_i
 
 	return nil
 }
+
+// IsControllerNode returns true if the supplied nodeID is a controller node.
+func (st *State) IsControllerNode(ctx context.Context, nodeID string) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	controllerNode := dbControllerNode{ControllerID: nodeID}
+
+	stmt, err := st.Prepare(`
+SELECT controller_id AS &dbControllerNode.*
+FROM controller_node
+WHERE controller_id = $dbControllerNode.controller_id`, controllerNode)
+	if err != nil {
+		return false, errors.Errorf("preparing select controller node statement: %w", err)
+	}
+
+	var result dbControllerNode
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, controllerNode).Get(&result)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("selecting controller node %q: %w", nodeID, err)
+		}
+		return nil
+	})
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+	return result.ControllerID == nodeID, nil
+}
```

Comment:
This is weird and confusing.

We pull out `controller_id` where `controller_id = $nodeID` and then return the bool `result.ControllerID == nodeID`.

It's feels very tautological.

I would rather we query a `COUNT(*)` and return `result.Count != 0`. This would be much clearer

Reply:
Yeah I agree with Jack, we already do the `COUNT` pattern on other places where we need to check for some row existence.